src/libraries/iKin/src/iKinFwd.cpp

```diff
@@ -1566,18 +1566,45 @@ iCubTorso::iCubTorso(const string &_type) : iKinLimb(string(""))
 /************************************************************************/
 void iCubTorso::allocate(const string &_type)
 {
+    double version;
+    size_t underscore = getType().find('_');
+    if (underscore != string::npos)
+    {
+        version = strtod(getType().substr(underscore + 2).c_str(), NULL);
+    }
+    else
+    {
+        version = 1.0;
+    }
+
     Matrix H0(4,4);
     H0.zero();
+    H0(3, 3) = 1.0;
+
+    if (version < 3.0)
+    {
     H0(0,1)=-1.0;
     H0(1,2)=-1.0;
     H0(2,0)=1.0;
-    H0(3,3)=1.0;
     setH0(H0);
 
     pushLink(new iKinLink(  0.032,     0.0,  M_PI/2.0,       0.0, -22.0*CTRL_DEG2RAD, 84.0*CTRL_DEG2RAD));
     pushLink(new iKinLink(    0.0, -0.0055,  M_PI/2.0, -M_PI/2.0, -39.0*CTRL_DEG2RAD, 39.0*CTRL_DEG2RAD));
     pushLink(new iKinLink(0.00231, -0.1933, -M_PI/2.0, -M_PI/2.0, -59.0*CTRL_DEG2RAD, 59.0*CTRL_DEG2RAD));
 }
+    else // version >= 3.0
+    {
+        H0(0, 2) = 1.0;
+        H0(1, 1) = -1.0;
+        H0(2, 0) = 1.0;
+        setH0(H0);
+
+        const double mm2m = 0.001;
+        pushLink(new iKinLink(72.5*mm2m, 0.0*mm2m, -90.0*CTRL_DEG2RAD, 0.0*CTRL_DEG2RAD));
```

Comment:
Please revise in order not to use `mm2m` for being consistent with the rest of the file.